Detectors/ITSMFT/ITS/simulation/src/V3Cage.cxx

```diff
@@ -0,0 +1,272 @@
+  //
+  // Output:
+  //
+  // Return:
+  //         The cover as a TGeoVolume
+  //
+  // Created:      22 Apr 2022  Mario Sitta
+  //
+
+  // Local variables
+  static const Int_t coverFoldNVert = 6;
+
+  Double_t xvert[coverFoldNVert], yvert[coverFoldNVert];
+  Double_t halfBase, zlen, alpha;
+
+  zlen = 0.5 * sCageCoverZLength;
+
+  // The cover core: a TGeoTubeSeg
+  halfBase = 0.5 * sCageCoverXBaseInt - sCageCoverSheetThick;
+  alpha = TMath::ACos(halfBase / sCageCoverCoreRext) * TMath::RadToDeg();
+  TGeoTubeSeg* coreSh = new TGeoTubeSeg("coverCore", sCageCoverCoreRint, sCageCoverCoreRext, zlen, alpha, 180. - alpha);
+
+  // The upper and lower sheets: a TGeoTubeSeg
+  // (as a whole volume - will become a sheet when the core is inserted inside)
+  halfBase = 0.5 * sCageCoverXBaseInt;
+  alpha = TMath::ACos(halfBase / sCageCoverRint) * TMath::RadToDeg();
+  TGeoTubeSeg* sheet = new TGeoTubeSeg("coverSheet", sCageCoverRint, sCageCoverCoreRext, zlen, alpha, 180. - alpha);
+
+  // The lateral fold: a Xtru
+  xvert[0] = sCageCoverXBaseExt / 2.;
+  yvert[0] = sCageCoverYBaseHeight + sCageCoverSheetThick;
+  xvert[1] = sCageCoverXWidth / 2.;
+  yvert[1] = yvert[0];
+  xvert[2] = xvert[1];
+  yvert[2] = sCageCoverYBaseHeight;
+  xvert[3] = sCageCoverXBaseInt / 2.;
+  yvert[3] = yvert[2];
+  xvert[4] = xvert[3];
+  yvert[4] = TMath::Sqrt(sCageCoverRint * sCageCoverRint - sCageCoverXBaseInt * sCageCoverXBaseInt / 4) + sCageCoverSheetThick;
+  xvert[5] = xvert[0];
+  yvert[5] = yvert[4];
+
+  TGeoXtru* fold = new TGeoXtru(2);
+  fold->SetName("coverFold");
+  fold->DefinePolygon(coverFoldNVert, xvert, yvert);
+  fold->DefineSection(0, -zlen);
+  fold->DefineSection(1, zlen);
+
+  // A BBox to cut away the curved portion above the fold
+  TGeoBBox* cutfold = new TGeoBBox("cutFold", xvert[1] - xvert[0], yvert[5] - yvert[1], 1.05 * zlen);
+
+  // Some matrices to create the composite shape
+  TGeoRotation* rotfold = new TGeoRotation("rotFold", 180, 180, 0);
+  rotfold->RegisterYourself();
+
+  TGeoTranslation* cutbox1 = new TGeoTranslation("cutBox1", xvert[1], yvert[5], 0);
+  cutbox1->RegisterYourself();
+
+  TGeoTranslation* cutbox2 = new TGeoTranslation("cutBox2", -xvert[1], yvert[5], 0);
+  cutbox2->RegisterYourself();
+
+  // The cover shape: a CompositeShape
+  TGeoCompositeShape* coverSh = new TGeoCompositeShape("coverSheet-cutFold:cutBox1-cutFold:cutBox2+coverFold+coverFold:rotFold");
+
+  // We have all shapes: now create the real volumes
+  TGeoMedium* medRohacell = mgr->GetMedium("ITS_ROHACELL$");
+  TGeoMedium* medPrepreg = mgr->GetMedium("ITS_M46J6K$");
+
+  TGeoVolume* coverVol = new TGeoVolume("CageCover", coverSh, medPrepreg);
+  coverVol->SetFillColor(kBlue);
+  coverVol->SetLineColor(kBlue);
+
+  TGeoVolume* coverCoreVol = new TGeoVolume("CageCoverCore", coreSh, medRohacell);
+  coverCoreVol->SetFillColor(kYellow);
+  coverCoreVol->SetLineColor(kYellow);
+
+  coverVol->AddNode(coverCoreVol, 1, 0);
```

Comment:
Hi @mario6829, this line makes fullCI failing.
```/sw/SOURCES/O2/8693-slc8_x86-64/0/Detectors/ITSMFT/ITS/simulation/src/V3Cage.cxx:207:38: error: use nullptr [modernize-use-nullptr]```

can you change the 0 with a `nullptr`?

Reply:
Hi @mconcas thanks for pointing it out! Fixed